File: src/Entity/Reclamation.php
<?php

namespace App\Entity;

use App\Repository\ReclamationRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ReclamationRepository::class)]
class Reclamation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;
    
    #[ORM\Column(length: 500, nullable: false)]
    #[Assert\Regex(
        pattern:"/^[a-zA-Z\s]+$/",
        message:"Le nom ne doit pas contenir de chiffres ou de symboles"
    )]
    private ?string $nom = null;
    
    
    #[Assert\Email(message : "The email '{{ value }}' is not a valid email.")]
    #[ORM\Column(length: 60, nullable: false)]
    private ?string $email = null;

    #[ORM\Column(length: 500, nullable: true)]
    private ?string $description = null;

    #[ORM\Column(nullable: true)]
    private ?bool $etat = false;

    #[ORM\ManyToOne(inversedBy: 'reclamations')]
    private ?Categorie $categorie = null;
    


    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(?string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(?string $email): self
    {
        $this->email = $email;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function isEtat(): ?bool
    {
        return $this->etat;
    }

    public function setEtat(?bool $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getCategorie(): ?Categorie
    {
        return $this->categorie;
    }

    public function setCategorie(?Categorie $categorie): self
    {
        $this->categorie = $categorie;

        return $this;
    }

    
}

File: src/entity/Reclamation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entity;

import java.util.Date;

public class Reclamation {


     private int id;
    private String nom;
     private String email;
      private String description;
    private boolean etat;
  
    public Reclamation() {
    }

    public Reclamation(String email, String objet, String description, int date_reclamation,  Boolean etat) {
        this.email = email;
        this.nom = objet;
        this.description = description;

    
        this.etat = etat;
    }

    public Reclamation(int id, String email, String objet, String description, int date_reclamation, Boolean etat) {
        this.id = id;
        this.email = email;
        this.nom = objet;
        this.description = description;
    
   
        this.etat = etat;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }
     public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }


    /**
     * @return the depot
     */
    public boolean getEtat() {
        return etat;
    }

    /**
     * @param depot the depot to set
     */
    public void setEtat(boolean etat) {
        this.etat = etat;
    }
    


    @Override
    public String toString() {
        return "Reclamation{" + "id=" + id + ", email=" + email + ", nom=" + nom + ", description=" + description   +  ", etat=" + etat + '}'+"\n";
    }
    
    
}
